middleware/authentication.js

```diff
@@ -0,0 +1,24 @@
+//const User = require('../models/User')
+const jwt = require('jsonwebtoken')
+const { UnauthenticatedError } = require('../errors')
+
+
+const authenticateUser = async (req, res, next) => {
+    const authHeader = req.headers.authorization
+    if (!authHeader && !authHeader.startsWith('Bearer')) {
+        throw new UnauthenticatedError('Authentication invalid')
+    }
+
+    const token = authHeader.split(' ')[1]
+    //TODO: does it make sense to make the model to the token validation (the same way it signs the token)
+    jwt.verify(token, process.env.JWT_SECRET, (error, decoded) => {
+        if (error) {
+            throw new UnauthenticatedError('Authentication invalid')
+        } else {
+            req.user = { userId: decoded.userId, name: decoded.name }
+            next()
+        }
+    })
+}
```

Comment:
I think yes -- `.verifyJWT` could be a method on the user model. However, on receiving an incoming request, I think you probably don't even want to query the database at all if you can avoid it. The beauty of a JWT is that we can cryptographically verify the token, so we can find out of the user is authenticated, who the user is, and whether the token is expired without needing to talk to the database at all.